.github/workflows/build.yml

```diff
@@ -0,0 +1,63 @@
+name: Build
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: ${{ matrix.board }}
+    container: golioth/golioth-zephyr-base:0.16.3-SDK-v0
+    env:
+      ZEPHYR_SDK_INSTALL_DIR: /opt/toolchains/zephyr-sdk-0.16.3
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Need to add xtensa-espressif_esp32s3_zephyr-elf toolchain to docker image
```

Comment:
Will make a follow-on PR after updating the docker image, but want to at least get this minimal CI in place ASAP.